albumentations/augmentations/dropout/coarse_dropout.py

```diff
@@ -129,39 +186,26 @@ def get_params_dependent_on_targets(self, params: Dict[str, Any]) -> Dict[str, A
         height, width = img.shape[:2]
 
         holes = []
-        for _ in range(random.randint(self.min_holes, self.max_holes)):
-            if all(
-                [
-                    isinstance(self.min_height, int),
-                    isinstance(self.min_width, int),
-                    isinstance(self.max_height, int),
-                    isinstance(self.max_width, int),
-                ]
-            ):
-                hole_height = random.randint(int(self.min_height), int(self.max_height))
-                hole_width = random.randint(int(self.min_width), int(self.max_width))
-            elif all(
-                [
-                    isinstance(self.min_height, float),
-                    isinstance(self.min_width, float),
-                    isinstance(self.max_height, float),
-                    isinstance(self.max_width, float),
-                ]
-            ):
-                hole_height = int(height * random.uniform(self.min_height, self.max_height))
-                hole_width = int(width * random.uniform(self.min_width, self.max_width))
+        num_holes_range: Tuple[int, int] = (
```

Comment:
No need to compute `num_holes_range` in `apply`.

It is already done in `InitSchema` in model_validator and all parameters code in `__init__` are already validated and preprocessed => `self.num_holes_range` and other ranges could be directly used here.

Reply:
`    def get_params_dependent_on_targets(self, params: Dict[str, Any]) -> Dict[str, Any]:
        img = params["image"]
        height, width = img.shape[:2]

        holes = []

        for _ in range(random.randint(*self.num_holes_range)):
            if all(isinstance(x, int) for x in self.hole_height_range + self.hole_width_range):
                hole_height = random.randint(*self.hole_height_range)
                hole_width = random.randint(*self.hole_width_range)
            elif all(isinstance(x, float) for x in self.hole_height_range + self.hole_width_range):
                hole_height = int(height * random.uniform(*self.hole_height_range))
                hole_width = int(width * random.uniform(*self.hole_width_range))
            else:
                msg = f"Hole width range and hole height range \
                    values should all either be ints or floats. \
                    Got: {[type(x) for x in self.hole_height_range + self.hole_width_range]} respectively"
                raise ValueError(msg)

            y1 = random.randint(0, height - hole_height)
            x1 = random.randint(0, width - hole_width)
            y2 = y1 + hole_height
            x2 = x1 + hole_width
            holes.append((x1, y1, x2, y2))

        return {"holes": holes}`

I did it like that in first time but mypy errors made me refactor

Reply:
Could you please share errors?
---
It is better to fix, even in hackish way mypy errors than code quality lower.
----
I suspect that if you do cast to Tuple in lines:

```python
self.num_holes_range = num_holes_range
self.hole_width_range = hole_width_range
self.hole_height_range = hole_height_range
```

say 
```
self.num_holes_range = cast(Tuple[int, int], num_holes_range)
```
the error will go away